services/src/unit-test/java/uk/gov/moj/sdt/services/utils/SdtBulkReferenceGeneratorTest.java

```diff
@@ -0,0 +1,76 @@
+package uk.gov.moj.sdt.services.utils;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import uk.gov.moj.sdt.dao.GenericDao;
+import uk.gov.moj.sdt.dao.api.IGenericDao;
+import uk.gov.moj.sdt.services.utils.api.ISdtBulkReferenceGenerator;
+import uk.gov.moj.sdt.utils.AbstractSdtUnitTestBase;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+
+
+@ExtendWith(MockitoExtension.class)
+public class SdtBulkReferenceGeneratorTest extends AbstractSdtUnitTestBase{
+    @Mock
+    IGenericDao genericDaoMock;
+
+    private static final String FOUR_CHARACTERS_ONLY ="The target application length is expected to be 4 characters.";
+
+    ISdtBulkReferenceGenerator sdtBulkReferenceGenerator;
+    @BeforeEach
+    @Override
+    public void setUp() {
+
+        sdtBulkReferenceGenerator = new SdtBulkReferenceGenerator(genericDaoMock);
+
+    }
+
+    @Test
+    void testGetSdtBulkReferenceGenerator() {
+
+        String generatedRef = sdtBulkReferenceGenerator.getSdtBulkReference("MCOL");
+
+        assertNotNull(generatedRef);
+    }
+
+    @Test
+    void testGetSdtBulkReferenceGeneratorIllegalArgumentException() {
+
+        try {
+                sdtBulkReferenceGenerator.getSdtBulkReference(null);
+        }catch(IllegalArgumentException ex){
+
+            assertEquals(FOUR_CHARACTERS_ONLY, ex.getMessage());
+        }
+
+    }
+
+    @Test
+    void testGetSdtBulkReferenceGeneratorTargetLengthIllegalArgumentException() {
+
+        try {
+             sdtBulkReferenceGenerator.getSdtBulkReference("over4Characters");
+        }catch(IllegalArgumentException ex){
+
+            assertEquals(FOUR_CHARACTERS_ONLY, ex.getMessage());
+        }
+    }
+
+    @Test
+    void testSetGenericDao() {
+
+        IGenericDao mockGenericDao = mock(GenericDao.class);
+        SdtBulkReferenceGenerator sdtBulkReferenceGeneratorObj = new SdtBulkReferenceGenerator(mockGenericDao);
+        sdtBulkReferenceGeneratorObj.setGenericDao(mockGenericDao);
+
+        assertNotNull(sdtBulkReferenceGeneratorObj,"Object should have been populated");
```

Comment:
Use the class field sdtBulkReferenceGenerator rather than creating a new one here.  Set it to a new mock GenericDao object then use assertEquals() to ensure that the set method worked correctly.

Reply:
unable ot do that as SdtBulkReferenceGenerator in the class field is Interface type.
I have called the abstract test method to check the genericDao has been set

Reply:
Change the type of the sdtBulkReferenceGenerator class field from ISdtBulkReferenceGenerator to SdtBulkReferenceGenerator.  This will then allow it to be used in this test.  We're testing the object rather than the interface it implements so this is an acceptable change.

Reply:
done